File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        boolean[] arr = new boolean[nums.length];
        
        for(int i : nums){
            if(arr[i]){
                return i;
            } else{
                arr[i] = true;
            }
        }
        return -1;
    }
}

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        vector<bool> exists(nums.size(), false);
        
        for(int num : nums){
            if(exists[num]){
                return num;
            } else{
                exists[num] = true;
            }
        }
        
        return -1;
    }
};